instrumentation.go

```diff
@@ -119,9 +133,46 @@ func newInstConfig(opts []InstrumentationOption) instConfig {
 }
 
 func (c instConfig) applyEnv() instConfig {
+	c = c.applyResourceAtrrEnv()
 	if v, ok := os.LookupEnv(envTargetExeKey); ok {
 		c.target = &process.TargetArgs{ExePath: v}
 	}
+	if v, ok := os.LookupEnv(envServiceNameKey); ok {
+		c.serviceName = v
+	} else if c.serviceName == "" {
+		c = c.setDefualtServiceName()
```

Comment:
I'm not sure we want to do `c = c.applyResourceAtrrEnv()` twice.
I thought of doing it only once in the start of the function (assuming we'll add support for more attributes in the future)

Reply:
Right. Updated the suggestion, that was a typo.

Reply:
I still think the `c = c.applyResourceAtrrEnv()` should come at the start of the function and not in the else block because it may include other attributes in the future.

Reply:
How would you propose we don't do wasted work then when it is going to be overwritten by the service name environment variable?

Reply:
If we want to support future attributes in OTEL_RESOURCE_ATTRIBUTES, I don't see an easy way to not parse it before looking for the more specific env variables.

Reply:
The way I've proposed resolves the immediate problem of avoiding duplicate work.

What do you think about resolving parsing OTEL_RESOURCE_ATTRIBUTES multiple times when we need to parse it multiple times? Maybe even just flatten this method into `applyEnv` so there is only one place this is done?

Reply:
Potentially parsing OTEL_RESOURCE_ATTRIBUTES multiple times seems to be the opposite of the goal - to not do wasted work.
In general, `newInstConfig` and `applyEnv` take the same approach of starting from the options with the lowest precedence and overriding them if necessary - which is inherently doing wasted work but simple code.

Reply:
I'm open to other suggestions on how to resolve the immediate problem of doing wasted work, like I said.

I'm not really seeing the point of opposition here. It sounds like you are suggesting we keep incorrect behavior now so we don't potentially do something wrong in the future. That doesn't play out in my understanding.